storage/index.go

```diff
@@ -365,7 +367,12 @@ func (i *nsIndex) writeBatchForBlockStartWithRLock(
 		// repeated inserts. as long as a block has an ID, it's not an error so we don't need
 		// to pollute the logs with these messages.
 		if partialError, ok := err.(*m3ninxindex.BatchPartialError); ok {
-			err = partialError.FilterDuplicateIDErrors()
+			partialError = partialError.FilterDuplicateIDErrors()
+			if partialError == nil {
+				err = nil
+			} else {
+				err = partialError
```

Comment:
Hm curious, why the if/else here?  If it's not it sets err to itself, if nil it explicitly sets err to nil - isn't the old way and this way equivalent?  (I'm sure its not, maybe we need to add a comment here?)

Reply:
Yeah it's weird. Arsies due to how interprets the return type of the Filter method. Made https://github.com/m3db/m3ninx/pull/83/files to avoid